From 8f0724a7a75680200dbb1d495a08a11bbbe4bcce Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Wed, 22 Feb 2017 11:38:46 -0800
Subject: [PATCH 2297/2748] Revert "HDFS-8307. Spurious DNS Queries from hdfs
 shell. Contributed by  Andres Perez"

This reverts commit 2d5f476f5e7556e6091ed69acfaba3411588b840.

Change-Id: Ic95e80484c8d112f2aa4ca237c60cba005a50d93
---
 .../org/apache/hadoop/hdfs/NameNodeProxies.java    |   14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
index d0732d7..d535de0 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
@@ -192,18 +192,8 @@ public InetSocketAddress getAddress() {
         dtService = SecurityUtil.buildTokenService(
             NameNode.getAddress(nameNodeUri));
       }
-
-      InetSocketAddress nnAddress;
-
-      //We dont need to resolve the address if is it a Nameservice ID
-      if(HAUtil.isLogicalUri(conf, nameNodeUri)) {
-        nnAddress = InetSocketAddress.createUnresolved(
-            nameNodeUri.getHost(), NameNode.DEFAULT_PORT);
-      } else {
-        nnAddress = NameNode.getAddressCheckLogical(conf, nameNodeUri);
-      }
-
-      return new ProxyAndInfo<T>(proxy, dtService, nnAddress);
+      return new ProxyAndInfo<T>(proxy, dtService,
+          NameNode.getAddressCheckLogical(conf, nameNodeUri));
     }
   }
   
-- 
1.7.9.5

